genealogos/src/backend/nixtract.rs

```diff
@@ -89,6 +91,62 @@ where
                     rev: src.rev.clone(),
                 });
 
+                // Convert the narinfo field of the DerivationDescription into a properties hashmap
+                let properties = {
+                    let mut acc = std::collections::HashMap::new();
+                    if let Some(narinfo) = &entry.nar_info {
+                        // These macros add the provided fields to the acc map. The first macro works with basic types. The second works with Options. The final one works with Vec<impl Display>.
+                        // TODO: Consider if this should be a Trait instead
+                        macro_rules! insert_into_map {
+                            ($($key:ident),+) => {
+                                $(
+                                    acc.insert(Some(format!("nix:narinfo:{}", stringify!($key))), Some(narinfo.$key.clone().to_string()));
+                                )+
+                            }
+                        }
+
+                        macro_rules! insert_into_map_option {
+                            ($($key:ident),+) => {
+                                $(
+                                    acc.insert(Some(format!("nix:narinfo:{}", stringify!($key))), narinfo.$key.clone().map(|v| v.to_string()));
+                                )+
+                            }
+                        }
+
+                        macro_rules! insert_into_map_vec {
+                            ($($key:ident),+) => {
+                                $(
+                                    acc.insert(Some(format!("nix:narinfo:{}", stringify!($key))), narinfo.$key.clone().map(|v| v.join(" ")));
+                                )+
+                            }
+                        }
+
+                        insert_into_map!(
```

Comment:
I find macros to be slightly overkill/heavy for this use case. But the a non-macro solution is also mildly annoying/verbose, and what those macros do is pretty clear, so both choices are reasonable, I guess.

Reply:
Looking back, I agree with you.

I've decided to turn it into a `HashMap::from()` that takes a slice of tuples.